### PR TITLE
feature/DISCOVERER (Final proto)

### PR DESCRIPTION
**Pull request by buccarel**
This is the final prototype of the Discoverer. It includes both the PaaSify and the CloudHarmony crawlers. Everything is accessible as RESTful service.
@szenzaro , since you are the one who uses the discoverer most, could you please review it?
Thanks.

### CODE REVIEW

File: discoverer/pom.xml [hunk 1]
```diff
@@ -32,26 +33,41 @@
...
 			<groupId>com.googlecode.json-simple</groupId>
 			<artifactId>json-simple</artifactId>
+			<version>1.1.1</version>
 		</dependency>
+
+		<!-- test units -->
 		<dependency>
 			<groupId>org.testng</groupId>
 			<artifactId>testng</artifactId>
+			<version>${testng.version}</version>
```
Comment (szenzaro) [inline discoverer/pom.xml]: you can remove the version inheriting it from the father

File: discoverer/pom.xml [hunk 2]
```diff
@@ -32,26 +33,41 @@
...
+
+		<dependency>
+			<groupId>eu.seaclouds-project</groupId>
+			<artifactId>common</artifactId>
+			<version>0.8.0-SNAPSHOT</version>
+		</dependency>
+
+		<!-- JSON parser -->
 		<dependency>
 			<groupId>com.googlecode.json-simple</groupId>
```
Comment (szenzaro) [inline discoverer/pom.xml]: this will probably be replaced by jackson.

**Comment by szenzaro** [comment]
Some tests would be nice but overall it's a +1.

**Comment by perezp** [comment]
Thank you for the PR, 

I agree with Simone that more test of the crawler should be provided.
Moreover, there are many defined Strings mixed the code, for example to define TOSCA yaml. 

It also is carrying some debt from the previous PR of the Discoverer. There is commented code and a folder "offerings_repo" is created in the user HOME. 

I merge it because we need to have this code inside and because it only modifies java files local to the Discoverer module, but we can undo the merge upon request.

**Pull request merged**